pkg/collector/corechecks/cluster/helm/helm.go

```diff
@@ -305,8 +279,24 @@ func (hc *HelmCheck) deleteSecret(obj interface{}) {
 	hc.deleteRelease(string(secret.Data["release"]), k8sSecrets)
 }
 
-func (hc *HelmCheck) updateSecret(_, obj interface{}) {
-	hc.addSecret(obj)
+func (hc *HelmCheck) updateSecret(old, new interface{}) {
+	oldSecret, ok := old.(*v1.Secret)
```

Comment:
updateSecret and updateConfigMap can be more generic if you convert the object to `metav1.ObjectMeta`

Reply:
The type assertion wouldn't work in that case
```
panic: interface conversion: interface {} is *v1.Secret, not v1.ObjectMeta [recovered]
	panic: interface conversion: interface {} is *v1.Secret, not v1.ObjectMeta
```